src/org/project/ProductManager.java

```diff
@@ -40,6 +41,31 @@ public Product getProductById(UUID id) {
         return products.stream()
                 .filter(product -> product.getId().equals(id))
                 .findFirst()
-                .orElseThrow(() -> new ProductNotFoundException("Produkt o ID " + id + " nie został znaleziony"));
+                .orElseThrow(() -> new ProductNotFoundException("Product with id " + id + " has not been found."));
+    }
+
+    public void addProductToCart(UUID productId, int quantity) {
+        Product product = getProductById(productId);
+        if (product.getQuantity() < quantity) {
+            throw new IllegalArgumentException("There is not enough product in stock.");
+        }
+        cart.addProductToCart(product.clone());
```

Comment:
tu na pewno jest blad, powinien wyjsc w testach jesli bysny testowali oba przypadki, ale widze jeden test wiec chyba nie testujemy.
w magazynie glwonym zmniejszamy ilosc w produkcie o zadane quantity, ale po stronie karty zwiekszamy ilosc o 1 zawsze, to odbiera sens tego quantity, imo powinnismy je przekazac do metody w cart i zwiekszyc ilosc tez tam o zadane quantity.